### ISSUE TITLE
[Rust] Implement new Concept Exercise: unit-structs

### ISSUE THREAD

**Issue by efx** [exercism/rust#1068]
This issue describes how to implement the `unit-structs` concept exercise for the rust track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal is to teach the student what unit structs are and how one can use them. You'll also see them referred to as zero-width structs.

## Learning objectives

- know the unit struct type
- know possible uses 

## Out of scope

- tuple structs
- structs

## Concepts

- unit-structs

## Prerequisites

- tuple structs
- structs

### Hints

- https://doc.rust-lang.org/std/primitive.unit.html

### After

- https://doc.rust-lang.org/std/primitive.unit.html

## Representer

No changes required

## Analyzer

No changes required

## Implementing

See our [implementation guide](https://github.com/exercism/v3/blob/master/languages/rust/howto/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise) for guidance.  In addition you may want to make use of the shell script while in the `concept` directory, with the arg loops, e.g. `sh boil_exercise.sh vec`

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
